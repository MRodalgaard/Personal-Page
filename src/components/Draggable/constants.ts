export const DRAGGABLE_HANDLE = 'handle';
