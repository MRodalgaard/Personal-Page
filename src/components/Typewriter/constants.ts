export const TYPEWRITER_DELAY = 30;
export const TYPEWRITER_OPACITY = 0.06;
