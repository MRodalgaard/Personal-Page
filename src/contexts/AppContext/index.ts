export { AppContextProvider } from './AppContextProvider';
export { useAppContext } from './useAppContext';
