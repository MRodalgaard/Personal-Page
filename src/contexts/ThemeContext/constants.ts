export type AppTheme = 'light' | 'dark';
export const PHONE_SIZE_PX = 500;
