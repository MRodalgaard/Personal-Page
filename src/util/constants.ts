// Key for persisting app state in local storage
export const LOCALSTORAGE_KEY = "AppState";

export enum AppBackground {
  default,
  greyscale,
  colored
}
