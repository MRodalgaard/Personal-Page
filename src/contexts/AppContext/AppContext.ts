import { AppTheme } from 'contexts/ThemeContext';
import { createContext } from 'react';

export interface IAppContext {
  colorized: boolean;
  theme: AppTheme;
  toggleColorized: () => void;
  toggleTheme: () => void;
}

export const defaultValue: IAppContext = {
  colorized: false,
  theme: 'light',
  toggleColorized: () => {},
  toggleTheme: () => {},
};

export const AppContext = createContext<IAppContext>(defaultValue);
