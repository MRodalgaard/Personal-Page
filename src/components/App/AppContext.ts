import { createContext } from 'react';
import { AppBackground } from '../../util/constants';
import { AppColors } from '../../util/theme';

export interface IAppState {
  color: AppColors;
  background: AppBackground;
  toggleBackground?: () => void;
  toggleColor?: () => void;
}

export const initialState: IAppState = {
  background: AppBackground.greyscale,
  color: AppColors.primary,
};

const AppContext = createContext(initialState);

export default AppContext;
