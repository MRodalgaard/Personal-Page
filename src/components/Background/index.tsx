import backgroundColorsImg from 'assets/img/background-colors.jpg';
import backgroundLowImg from 'assets/img/background-low.jpg';
import backgroundImg from 'assets/img/background.jpg';
import AppContext from 'components/App/AppContext';
import React, { useContext, useEffect, useRef } from 'react';
import {
  ImageProps,
  LazyImage,
  LazyImageRenderPropArgs,
  RefArg,
} from 'react-lazy-images';
import styled from 'styled-components';
import { AppBackground } from 'util/constants';

const Img = styled.img`
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: -1;
  object-fit: cover;

  animation-duration: 4s;
  animation-fill-mode: both;
`;

const ImgFadeIn = styled(Img as any)`
  animation-name: fadeIn;

  @keyframes fadeIn {
    from {
      opacity: 0;
    }

    to {
      opacity: 1;
    }
  }
`;

const ImgFadeOut = styled(Img as any)`
  animation-name: fadeOut;

  @keyframes fadeOut {
    from {
      opacity: 1;
    }

    to {
      opacity: 0;
    }
  }
`;

const Background = () => {
  const { background } = useContext(AppContext);

  const firstUpdate = useRef(true);

  // Use update flag to accommodate for first greyscale background fade in
  useEffect(() => {
    const timeoutId = window.setTimeout(
      () => (firstUpdate.current = false),
      500
    );
    return () => {
      clearTimeout(timeoutId);
    };
  }, []);

  // Do not render background on phone size screens
  if (window.innerWidth < 500) {
    return null;
  }

  const placeholder = ({
    imageProps,
    ref,
  }: LazyImageRenderPropArgs & RefArg) => (
    <ImgFadeIn ref={ref} src={backgroundLowImg} alt={imageProps.alt} />
  );

  const actualFadeIn = ({ imageProps }: { imageProps: ImageProps }) => (
    <ImgFadeIn {...imageProps} />
  );

  const actualFadeOut = ({ imageProps }: { imageProps: ImageProps }) => (
    <ImgFadeOut
      hidden={firstUpdate.current && background === AppBackground.greyscale}
      {...imageProps}
    />
  );

  return (
    <>
      <LazyImage
        src={backgroundImg}
        placeholder={placeholder}
        actual={actualFadeIn}
        alt="Background"
      />
      <LazyImage
        src={backgroundColorsImg}
        placeholder={placeholder}
        actual={
          background === AppBackground.colored ? actualFadeIn : actualFadeOut
        }
        alt="Background Color"
      />
    </>
  );
};

export default Background;
