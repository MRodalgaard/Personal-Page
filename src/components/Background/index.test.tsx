import { render } from '@testing-library/react';
import AppContext, { initialState } from 'components/App/AppContext';
import React from 'react';
import { AppBackground } from 'util/constants';
import Background from '.';

describe('Background', () => {
  it('renders', () => {
    const { container } = render(
      <AppContext.Provider value={initialState}>
        <Background />
      </AppContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders using other background', () => {
    const initialStateMock = {
      ...initialState,
      background: AppBackground.greyscale,
    };

    const { container } = render(
      <AppContext.Provider value={initialStateMock}>
        <Background />
      </AppContext.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
