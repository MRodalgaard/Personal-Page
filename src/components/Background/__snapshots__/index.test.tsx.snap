// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background renders 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: -1;
  object-fit: cover;
  -webkit-animation-duration: 4s;
  animation-duration: 4s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c1 {
  -webkit-animation-name: fadeIn;
  animation-name: fadeIn;
}

<img
  alt="Background"
  class="c0 c1"
  src="background-low.jpg"
/>
`;

exports[`Background renders using other background 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: -1;
  object-fit: cover;
  -webkit-animation-duration: 4s;
  animation-duration: 4s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.c1 {
  -webkit-animation-name: fadeIn;
  animation-name: fadeIn;
}

<img
  alt="Background"
  class="c0 c1"
  src="background-low.jpg"
/>
`;
